backend/worker/quivr_worker/celery_worker.py

```diff
@@ -129,8 +176,12 @@ async def process_crawl_task(
     logger.info(
         f"Task process_crawl_task started for url={crawl_website_url}, knowledge_id={knowledge_id}, brain_id={brain_id}, notification_id={notification_id}"
     )
-
-    brain_vector_service = BrainVectorService(brain_id)
+    global engine
+    assert engine
+    with Session(engine, expire_on_commit=False, autoflush=False) as session:
+        brain_vector_service = BrainVectorService(brain_id)
+        vector_repository = VectorRepository(session)
+        vector_service = VectorService(vector_repository)
```

Comment:
Beware of context manager scope. The vector_service isn't in the scope for process_url_func